assignment 6A/IntSet.java

```diff
@@ -0,0 +1,140 @@
+package Assignment6A;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class represent immutable abstract IntSet data type
+ * @author Sabir
+ *
+ */
+public class IntSet {
+
+	private final int MAX_SET_SIZE = 1000;
+	private final int integerArray[];
+
+	/**
+	 * Parameterized constructor used to initialize instance variable integer array
+	 * @param integerArray  integer single dimension array containing integer between 1 to 1000
+	 */
+	public IntSet(int integerArray[]) {
+		this.integerArray = integerArray;
+	}
+
+	/**
+	 * This method check if given integer value is member of the set
+	 * @param value integer value between 1 to 1000
+	 * @return true if integer value is part of set else false
+	 */
+	public boolean isMember(int value) {
+		for (int index = 0; index < this.integerArray.length; index++) {
+			if (this.integerArray[index] == value) { return true; }
+		}
+		return false;
+	}
+
+	/**
+	 * This method find size of the integer set
+	 * @return integer representing size of the set
+	 */
+	public int size() {
+		return this.integerArray.length;
+	}
+
+	/**
+	 * This method check if given parameter set is subset or not
+	 * @param set integer set containing integer between 1 to 1000
+	 * @return true if given set is subset else false
+	 */
+	public boolean isSubSet(IntSet set) {
+
+		boolean isSet;
+		for (int firstIndex = 0; firstIndex < set.integerArray.length; firstIndex++) {
+			isSet = false;
+			for (int secondIndex = 0; secondIndex < this.integerArray.length; secondIndex++) {
+				if (set.integerArray[firstIndex] == this.integerArray[secondIndex]) {
+					isSet = true;

```

Comment:
You can optimize execution here by using break;

Reply:
yes sir i will try to optimize my logic more